pkg/util/ec2/ec2_tags.go

```diff
@@ -66,6 +77,10 @@ func fetchEc2TagsFromIMDS(ctx context.Context) ([]string, error) {
 		if err != nil {
 			return nil, err
 		}
+		if isTagExcluded(key) {
```

Comment:
Do we want to make the call to `getMetadataItem` if we know that the tag is already exclude ?

Reply:
`getMetadataItem` is used for more than just tags (network id, ipv4, ...).